internal/font_metadata.py

```diff
@@ -0,0 +1,120 @@
+# Copyright 2020 WebPageTest LLC.
+# Copyright 2020 Google Inc.
+# Use of this source code is governed by the Apache 2.0 license that can be
+# found in the LICENSE file.
+"""Extract metadata from OpenType fonts."""
+
+from fontTools.ttLib import TTFont
+import functools
+import logging
+
+
+_NAME_ID_VERSION = 5
+_NAME_ID_POSTSCRIPT_NAME = 6
+_NAME_ID_LICENSE_URL = 14
+_MAX_NAME_LEN = 64
+
+
+def _read_names(ttf, name_ids):
+    names = {}
+    try:
+        for name in ttf['name'].names:
+            if not name.nameID in name_ids:
+                continue
+            if not name.isUnicode():
+                continue
+            try:
+                names[name.nameID] = name.toUnicode()[:_MAX_NAME_LEN]
+            except Exception:
+                logging.exception('Error converting name to unicode')
+
+    except Exception:
+        logging.exception('Error reading font names')
+    if not names:
+        return None
+    return names
+
+
+def _read_os2(ttf):
+    try:
+        os2 = ttf['OS/2']
+        return {
+            'usWeightClass': os2.usWeightClass,
+            'usWidthClass': os2.usWidthClass,
```

Comment:
I'd like to get more os2 table values :)

```
fsType
panose
achVendID
sxHeight  
sCapHeight
sDefaultChar
usLowerOpticalPointSize
usUpperOpticalPointSize
ulUnicodeRange1
ulUnicodeRange2
ulUnicodeRange3
ulUnicodeRange4
ulCodePageRange1
ulCodePageRange2
```

Reply:
Would it make sense to just return all of them? It doesn't look like it will bloat the result too badly.

Reply:
> Would it make sense to just return all of them? It doesn't look like it will bloat the result too badly.

Sure, if its cheaper and easier to just get them all. I don't have any preconceived ideas for the keys I left out, but maybe in future someone will come up with some :)

Reply:
Similarly, these `ttf['post']` keys can be interesting, and if getting all is cheap, might make sense to get all for `post`, `name`, `OS/2` and `STAT`. 

Type | Name | Description
-- | -- | --
Fixed | version | 0x00010000 for version 1.00x00020000 for version 2.00x00025000 for version 2.5 (deprecated)0x00030000 for version 3.0
Fixed | italicAngle | Italic angle in counter-clockwise degrees from the vertical. Zero for upright text, negative for text that leans to the right (forward).
FWord | underlinePosition | This is the suggested distance of the top of the underline from the baseline (negative values indicate below baseline).The PostScript definition of this FontInfo dictionary key (the y coordinate of the center of the stroke) is not used for historical reasons. The value of the PostScript key may be calculated by subtracting half the underlineThickness from the value of this field.
FWord | underlineThickness | Suggested values for the underline thickness. In general, the underline thickness should match the thickness of the underscore character (U+005F LOW LINE), and should also match the strikeout thickness, which is specified in the OS/2 table.
uint32 | isFixedPitch | Set to 0 if the font is proportionally spaced, non-zero if the font is not proportionally spaced (i.e. monospaced).

Reply:
Tables that have dynamic size is potentially problematic unless we add code to limit the max possible size. Fetching all of OS/2 is OK on grounds of fixed size so I'll start with that.

Reply:
See #371.